test/quadiron_c_utest.cpp

```diff
@@ -0,0 +1,246 @@
+    {
+        std::fill_n(dst.begin(), n_dst, 0);
+        for (int i = 0; i < n_src; i++) {
+            ASSERT_LT(src[i], n_dst);
+            dst[src[i]] = 1;
+        }
+    }
+
+    /** Test encode/decode/reconstruct
+     *
+     * This test will create n_data fragments and generate n_parities parities,
+     * then decode with a set of missing fragments. Finaly reconstruct them.
+     *
+     * @param n_data number of data
+     * @param n_parities number of parities
+     * @param block_size size of block in bytes
+     * @param systematic 1 if systematic else 0
+     * @param missing_idxs vector of boolean vales indicating missing fragments
+     * for decode and reconstruct
+     *        - must be of length n_parities
+     */
+    void test_encode_decode_reconstruct(
+        int n_data,
+        int n_parities,
+        size_t block_size,
+        int systematic,
+        std::vector<int> missing_idxs)
+    {
+        struct QuadironFnt32* inst =
+            quadiron_fnt32_new(2, n_data, n_parities, systematic);
+        size_t metadata_size =
+            quadiron_fnt32_get_metadata_size(inst, block_size);
+        std::vector<std::vector<uint8_t>> data(n_data);
+        std::vector<uint8_t*> _data(n_data); // for C API
+        std::vector<std::vector<uint8_t>> ref_data(n_data);
+        std::vector<uint8_t*> _ref_data(n_data); // for C API
+        int n_outputs;
+        if (systematic) {
+            n_outputs = n_parities;
+        } else {
+            n_outputs = n_data + n_parities;
+        }
+        std::vector<std::vector<uint8_t>> parity(n_outputs);
+        std::vector<uint8_t*> _parity(n_outputs); // for C API
+        std::vector<int> wanted_idxs(n_outputs);
+
+        for (int i = 0; i < n_data; i++) {
+            data.at(i).resize(block_size + metadata_size);
+            _data[i] = data.at(i).data();
+            ref_data.at(i).resize(block_size);
+            _ref_data[i] = ref_data.at(i).data();
+            randomize_buffer(_data[i] + metadata_size, block_size);
+            std::memcpy(_ref_data[i], _data[i] + metadata_size, block_size);
+        }
+
+        for (int i = 0; i < n_outputs; i++) {
+            parity.at(i).resize(block_size + metadata_size);
+            _parity[i] = parity.at(i).data();
+        }
+
+        // we want all parities
+        for (int i = 0; i < n_outputs; i++) {
+            wanted_idxs[i] = 1;
+        }
+
+        ASSERT_EQ(
+            quadiron_fnt32_encode(
+                inst,
+                _data.data(),
+                _parity.data(),
+                wanted_idxs.data(),
+                block_size),
+            0);
+
+        for (int i = 0; i < n_data; i++) {
+            if (missing_idxs[i]) {
+                std::memset(_data[i], 0, block_size + metadata_size);
+            }
+        }
+
+        for (int i = 0; i < n_parities; i++) {
+            if (missing_idxs[n_data + i]) {
+                std::memset(_parity[i], 0, block_size + metadata_size);
```

Comment:
replace `std::memset` by `std::fill_n` (I tried locally and the tests still works, as expected.